content/docs/command-reference/exp/run.md

```diff
@@ -69,27 +69,27 @@ checkpoints with DVC during your code or script runtime (similar to a logger).
 
 To do so, first mark stage `outs` with `checkpoint: true` in `dvc.yaml`. Having
 at least a checkpoint <abbr>output</abbr> is needed so that the experiment can
-later restart based on that output's last <abbr>cached</abbr> state.
+later continue from that output's last <abbr>cached</abbr> state.
 
 ⚠️ Using the `checkpoint` field in `dvc.yaml` is only compatibly with
 `dvc exp run`, `dvc repro` will abort if any stage contains it.
 
-Then, in the corresponding code, either call the `dvc.api.make_checkpoint()`
-function (Python), or write a signal file (any programming language) following
-the same steps as `make_checkpoint()` — please refer to that reference for
-details.
+Then, in your code either call the `dvc.api.make_checkpoint()` function
+(Python), or write a signal file (any programming language) following the same
+steps as `make_checkpoint()` — please refer to its reference for details.
 
-Once this is setup, you can use `dvc exp run` to begin the experiment. When the
-process finishes or gets interrupted (e.g. with Ctrl + `C`), DVC will
-[apply](/doc/command-reference/exp/apply) the last checkpoint to the
-<abbr>workspace</abbr> (overwriting any further changes done by the stage).
+You can now use `dvc exp run` to begin the experiment. This removes any
+`checkpoint` outputs before running the experiment (regardless of whether they
+have cached versions). When the process finishes or gets interrupted (e.g. with
+Ctrl + `C`), DVC will [apply](/doc/command-reference/exp/apply) the last
+checkpoint to the <abbr>workspace</abbr> (overwriting any further changes done
+by the stage).
```

Comment:
I think wording it this way might end up confusing users?

>  This removes any `checkpoint` outputs before running the experiment (regardless of whether they have cached versions).

`checkpoint` outs are only removed before running the experiment if `--reset` is specified. In a normal `dvc exp run` with no other arguments, we continue the experiment using the most recent version of that output.

If there is no previous run at all and the output has never existed before, then yes `dvc exp run` will do the same thing as `dvc exp run --reset`, but only in that specific case.

Reply:
Can this sentence get deleted? There's nothing special about what happens when you begin a checkpoint experiment. It's only different after at least one checkpoint has been generated.

Reply:
> There's nothing special about what happens when you begin a checkpoint experiment

What about what Peter just mentioned @dberenbaum? "If there is no previous run at all and the output has never existed before, then yes dvc exp run will do the same thing as dvc exp run --reset" 🙂

Reply:
> wording it this way might end up confusing users?

@pmrowla good point but maybe the problem is that the behavior can be a bit confusing per-se. I'll see if there's a more explicit way to express this but it already states "use `dvc exp run` **to begin** the experiment. This removes any `checkpoint` outputs" (it specifies "to begin").

Reply:
That's not really a special case though, and even in that case nothing is being removed (since there is no initial data to even remove in the first place)

Reply:
Nvmd I think my latest update should make it very clear. It's getting a little long/convoluted but at least it's complete and correct (I think). See 9e1a7e94.

Reply:
> even in that case nothing is being removed (since there is no initial data to even remove in the first place

What if you just changed regular outputs into checkpoint outputs in the last commit? (to migrate from regular pipeline to a checkpoint experiment)

Reply:
So the updated sentence is:

```
On this very first use,
any `checkpoint` outputs are deleted before running the experiment (regardless
of whether they have <abbr>cached</abbr> versions). 
```

> It's getting a little long/convoluted but at least it's complete and correct (I think).

Agreed on all counts 😄 . I think I get your point now that it's unclear whether an initial checkpoint run treats existing outs as checkpoints or not. Ideally, this is called out as an edge case since it might add confusion for the typical case where no outs exist yet, but 🤷 .

Reply:
> the typical case where no outs exist yet

OK if its an edge case we can leave it undocumented. Always good to simplify docs. Removed that sentence for now... Not sure I fully understand all the possible scenarios but will try to QA separately.